x-pack/plugins/security_solution/public/timelines/containers/kpis/index.tsx

```diff
@@ -0,0 +1,132 @@
+  isBlankTimeline: boolean;
+}
+
+export const useTimelineKpis = ({
+  timerange,
+  filterQuery,
+  docValueFields,
+  defaultIndex,
+  isBlankTimeline,
+}: UseTimelineKpiProps): [boolean, TimelineKpiStrategyResponse | null] => {
+  const { data, notifications } = useKibana().services;
+  const refetch = useRef<inputsModel.Refetch>(noop);
+  const abortCtrl = useRef(new AbortController());
+  const [loading, setLoading] = useState(false);
+  const [timelineKpiRequest, setTimelineKpiRequest] = useState<TimelineRequestBasicOptions | null>(
+    null
+  );
+
+  const [
+    timelineKpiResponse,
+    setTimelineKpiResponse,
+  ] = useState<TimelineKpiStrategyResponse | null>(null);
+
+  const timelineKpiSearch = useCallback(
+    (request: TimelineRequestBasicOptions | null) => {
+      if (request == null) {
+        return;
+      }
+
+      let didCancel = false;
+      const asyncSearch = async () => {
+        abortCtrl.current = new AbortController();
+        setLoading(true);
+
+        const searchSubscription$ = data.search
+          .search<TimelineRequestBasicOptions, TimelineKpiStrategyResponse>(request, {
+            strategy: 'securitySolutionTimelineSearchStrategy',
+            abortSignal: abortCtrl.current.signal,
+          })
+          .subscribe({
+            next: (response) => {
+              if (isCompleteResponse(response)) {
+                if (!didCancel) {
+                  setLoading(false);
+                  setTimelineKpiResponse(response);
+                }
+                searchSubscription$.unsubscribe();
+              } else if (isErrorResponse(response)) {
+                if (!didCancel) {
+                  setLoading(false);
+                }
+                notifications.toasts.addWarning('An error has occurred');
+                searchSubscription$.unsubscribe();
+              }
+            },
+            error: (msg) => {
+              if (!didCancel) {
+                setLoading(false);
+              }
+              if (!(msg instanceof AbortError)) {
+                notifications.toasts.addDanger('Failed to load KPIs');
+              }
+            },
+          });
+      };
+      abortCtrl.current.abort();
+      asyncSearch();
+      refetch.current = asyncSearch;
+      return () => {
+        didCancel = true;
+        abortCtrl.current.abort();
+      };
+    },
+    [data.search, notifications.toasts]
+  );
+
+  useEffect(() => {
+    setTimelineKpiRequest((prevRequest) => {
+      const myRequest = {
+        ...(prevRequest ?? {}),
+        docValueFields,
+        defaultIndex,
+        timerange,
+        filterQuery,
+        factoryQueryType: TimelineEventsQueries.kpi,
+      };
+      if (!deepEqual(prevRequest, myRequest)) {
+        return myRequest;
+      }
+      return prevRequest;
+    });
+  }, [docValueFields, defaultIndex, timerange, filterQuery]);
+
+  useEffect(() => {
+    if (isBlankTimeline) {
+      timelineKpiSearch(null);
```

Comment:
What's the point of making a request that'll just return null? Can you comment?

Reply:
this was an extra line from an earlier way of doing this that after some refactoring amounted to just a noop basically, good catch though removing.